### ISSUE TITLE
bun patch ask for version when only one is available

### ISSUE THREAD

**Issue by ArnaudBarre** [oven-sh/bun#12255]
### What version of Bun is running?

1.0.17

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

`bun add typescript ts-api-utils`
`bun patch typescript`

### What is the expected behavior?

patch proceed as expected

### What do you see instead?

```
error: Found multiple versions of typescript, please specify a precise version from the following list:
  typescript@5.5.2
```

### Additional information

I think this is because lockfile looks like this:
```
typescript@>=4.2.0, typescript@^5.5.2:
  version "5.5.2"
  resolved "https://registry.npmjs.org/typescript/-/typescript-5.5.2.tgz"
  integrity sha512-NcRtPEOsPFFWjobJEtfihkLCZCXZt/os3zf8nTxjVH3RvTSxjrCamJpbExGvYOF+tFHc3pA65qpdwPbzjohhew==
```

**Comment by dylan-conway** [comment]
this will be fixed with #12141

**Comment by dylan-conway** [comment]
Fixed in #12141 and will be included in bun `v1.1.18`
